server/controllers/response.ts

```diff
@@ -155,3 +156,44 @@ export function getAllResponse(surveyId: string, done: (err: any, res?: Response
         }
     });
 }
+export async function downloadRaw (req: Request, res: Response, next: NextFunction) {
+    if (!req.user) { // if req.user is undefined
+        throw Error("Unreachable: this route handler is called only when the user is logged in");
+    }
+    const id = req.params.id;
+    const questionsCol = await Survey.findById(id);
+    const fields =
+    [
+        {
+            label: questionsCol?.title,
+            value: "title",
+        },
+        {
+            label: questionsCol?.questions[0],
+            value: "answers[0]",
+        },
+        {
+            label: questionsCol?.questions[1],
+            value: "answers[1]",
+        },
+        {
+            label: questionsCol?.questions[2],
+            value: "answers[2]",
+        },
+        {
+            label: questionsCol?.questions[3],
+            value: "answers[3]",
+        },
+        {
+            label: questionsCol?.questions[4],
+            value: "answers[4]",
+        },
+        {
+            label: "Created At",
+            value: "createdAt",
+        },
+    ];
+    const data = await ResponseM.find({ question: id });
+    //@ts-ignore
+    downloadResource(res, "response.csv", fields, data);
```

Comment:
Your `downloadResource` method has the signature of `(res: Response, filename: string, fields: string[], data: object[]) => void`, but for the `fields` parameter, you're inputting this:

```ts
const fields =
    [
        {
            label: questionsCol?.title,
            value: "title",
        },
        {
            label: questionsCol?.questions[0],
            value: "answers[0]",
        },
        {
            label: questionsCol?.questions[1],
            value: "answers[1]",
        },
        {
            label: questionsCol?.questions[2],
            value: "answers[2]",
        },
        {
            label: questionsCol?.questions[3],
            value: "answers[3]",
        },
        {
            label: questionsCol?.questions[4],
            value: "answers[4]",
        },
        {
            label: "Created At",
            value: "createdAt",
        },
    ];
```

So, the input has `{ label: string | undefined, value: string }[]` instead of the required `string[]`. 

You have to change the type signature of your `downloadResource` function. I'll mark them with suggestions below.